### PR TITLE
Profiling tools

### LINKED ISSUE #123: New Content: Profiling tools
A complete overview of profiling tools could consist of the following items:

- MessageTally/TimeProfileBrowser: start profiler from the docking bar, <kbd>spy on it</kbd>, `BlockClosure>>#timeProfile`
- `#timeToRun`, `#bench[For:]`, <kbd>tally it</kbd>
- SpaceTally
- (advanced: inspect bytecode of a method to optimize inlining)

Currently, chapter 6.11 mentions the message tally briefly only.

As always, the general question is: Is this relevant and yes, in how many sections/pages should this be covered?

(Side note: Should we finally always use the TimeProfileBrowser for displaying MessageTally results?)

### PR DESCRIPTION
**Pull request by LinqLover**
Close #123

- [x] check new screenshots on CI (all sq versions)

### CODE REVIEW

File: Environment/Environment.tex [hunk 1]
```diff
@@ -1282,20 +1282,107 @@ \section{Other interesting tools}
...
+
+\begin{ExecuteSmalltalkScript}
+SBEScreenshotRecorder writeTo: './figures/TimeProfile.png' building: [:helper |
+	| sem window splitter list |
+	sem := Semaphore new.
+	[Processor activeProcess environmentAt: ActiveWorldVariable put: helper world. "avoid #useWorldAndHandDuring: to keep stack clean"
+	[World imageForm] timeProfile.
+	sem signal] fork.
+	sem wait.
+	Transcript showln: 'then'.
```
Comment (LinqLover) [inline Environment/Environment.tex:1306]:
```suggestion
```

File: Environment/Environment.tex [hunk 2]
```diff
@@ -1282,20 +1282,106 @@ \section{Other interesting tools}
...
+		\ifluluelse
+			{\includegraphics[width=\textwidth]{SpaceTally}}
+			{\includegraphics[scale=0.7]{exploreWithLabel: 'SpaceTally'}}
+		\label{fig:SpaceTally}
+	\end{minipage}
+	\caption{A time profiler (left) and a space tally (right).}
+\end{figure}
+
+If you have discovered that your code is too slow, you might wonder which parts of your code are responsible for the slowness.
+For that, \sq provides the \emphind{message tally} toool.
```
Comment (codeZeilen) [inline Environment/Environment.tex:1356]: toool -> tool

File: Environment/Environment.tex [hunk 3]
```diff
@@ -1282,20 +1282,106 @@ \section{Other interesting tools}
...
+You can start it to record the execution times for your code by selecting the code and choosing \menu{spy on it} from the context menu.
+Alternatively, you can let it observe all processes in the image from the menu item \menu{world main docking bar~\go{} Apps~\go{} Message Tally}.
+Note that the message tally is a sampling profiler, so for detailed information, you will have to let your code run for some seconds (e.g. by running it 1000 times if it only takes little time).
```
Comment (codeZeilen) [inline Environment/Environment.tex:1359]: vll: "little time" -> "short time"

File: Environment/Environment.tex (hunk 2 above)
Comment (LinqLover) [reply Environment/Environment.tex:1356]: I should fix my Grammarly extension 😅

File: Environment/Environment.tex (hunk 3 above)
Comment (LinqLover) [reply Environment/Environment.tex:1359]: According to DeepL Write, Google, and Grammarly, "a little time" (but I missed the "a") is actually more popular than "short time". :)

```suggestion
Note that the message tally is a sampling profiler, so for detailed information, you will have to let your code run for some seconds (e.g. by running it 1000 times if it only takes a little time).
```

File: Environment/Environment.tex [hunk 4]
```diff
@@ -1282,20 +1282,106 @@ \section{Other interesting tools}
...
+]
+\end{ExecuteSmalltalkScript}
+\begin{ExecuteSmalltalkScript}
+SBEScreenshotRecorder writeTo: './figures/SpaceTally.png' building: [:helper |
+	(SpaceTally new depth: 2; spaceTally: (PackageInfo named: #Morphic) classes) exploreWithLabel: 'SpaceTally'.
+	helper scaleWindow: helper foregroundWindow extent: 500 px @ 225 px.
+]
+\end{ExecuteSmalltalkScript}
+
+First, you can send the message \mthind{BlockClosure}{timeToRun} to a block to measure the time it takes to execute it:
```
Comment (LinqLover) [inline Environment/Environment.tex:1325]:
```suggestion
First, you can send the message \mthind{BlockClosure}{timeToRun} to a block to measure the time in milliseconds it takes to execute it:

```

File: Environment/Environment.tex [hunk 5]
```diff
@@ -1282,20 +1282,106 @@ \section{Other interesting tools}
...
+		\label{fig:SpaceTally}
+	\end{minipage}
+	\caption{A time profiler (left) and a space tally (right).}
+\end{figure}
+
+If you have discovered that your code is too slow, you might wonder which parts of your code are responsible for the slowness.
+For that, \sq provides the \emphind{message tally} tool.
+You can start it to record the execution times for your code by selecting the code and choosing \menu{spy on it} from the context menu.
+Alternatively, you can let it observe all processes in the image from the menu item \menu{world main docking bar~\go{} Apps~\go{} Message Tally}.
+Note that the message tally is a sampling profiler, so for detailed information, you will have to let your code run for some seconds (e.g. by running it 1000 times if it only takes a little time).
```
Comment (LinqLover) [inline Environment/Environment.tex:1359]:
```suggestion
Note that the message tally is a sampling profiler, so for detailed information, you will have to let your code run for some seconds (\eg by running it 1000 times if it only takes a little time).
```

**Pull request merged**